### PR TITLE
Added moniker insertion feature & unit test

### PR DESCRIPTION
**Pull request by JohnTheFarmer**
new files : moniker-controller.ts & moniker-controller.test.ts

### CODE REVIEW

**Comment by bharney0** [comment]
#362

**Comment by bharney0** [comment]
@JohnTheFarmer Thank you for the contribution! We are reviewing your Pull Request right now and will provide feedback based on your changes.

File: docs-markdown/src/controllers/moniker-controller.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
+"use strict";
+
+import { Range, TextEditor, window } from "vscode";
+import { insertContentToEditor, setCursorPosition, isMarkdownFileCheck, noActiveEditorMessage } from "../helper/common";
+import { isCursorInsideYamlHeader } from "../helper/yaml-metadata";
+import { sendTelemetryData } from "../helper/telemetry";
+
+const telemetryCommand: string = "insertMoniker";
+let sign: string = "";
```
Comment (bharney0) [inline docs-markdown/src/controllers/moniker-controller.ts]: Please declare this within the `insertMoniker` function to reduce global variables.

File: docs-markdown/src/controllers/moniker-controller.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        return;
+    }
+
+    //is cursor on a new line 
+    if (isContentOnCurrentLine(editor)) {
+        window.showErrorMessage("Moniker must be inserted on a new line.");
+        return;
+    }
+
+    const moniker_options = [
```
Comment (bharney0) [inline docs-markdown/src/controllers/moniker-controller.ts]: We try to keep the variable names as camelCase. Please change `moniker_options` to `monikerOptions`.

File: docs-markdown/src/controllers/moniker-controller.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        "range equals",
+        "range greater than or equal",
+        "range less than or equal",
+    ];
+
+    window.showQuickPick(moniker_options).then((qpSelection) => {
+        if (!qpSelection) {
+            return;
+        }
+        if (qpSelection == moniker_options[0]) {
```
Comment (bharney0) [inline docs-markdown/src/controllers/moniker-controller.ts]: Please use triple equals === for these if statements.

**Comment by bharney0** [comment]
@JohnTheFarmer The changes look good and seem to follow the same patterns that are being used in the other controllers. And thank you for including unit tests. There are some very small changes in the feedback.

**Comment by JohnTheFarmer** [comment]
@bharney0  I think its a bug [#issue145](https://github.com/microsoft/vscode-docs-authoring/issues/145#issue-416571511)

**Comment by bharney0** [comment]
@JohnTheFarmer we expect these changes to be released sometime this week. You can test your changes on docs-markdown version 0.2.61. The release will likely be this Thursday 04/02.

**Pull request merged**